Report config to complete json

dannyroberts opened the pull request:
http://manage.dimagi.com/default.asp?175848

dannyroberts committed:
rewrite ReportConfig.to_complete_json to be explicit

dannyroberts committed:
make ReportConfig.by_domain_and_owner explicit

dannyroberts committed:
skip over deleted reports in saved configs

dannyroberts commented:
@czue @TylerSheffels

TylerSheffels commented on corehq/apps/reports/views.py:
```diff
@@ -165,7 +165,16 @@ def saved_reports(request, domain, template="reports/reports_home.html"):
+        if config.is_configurable_report:
+            try:
+                config.configurable_report.spec
+            # todo: there's got to be a better thing to raise in this situation
```
I kind of agree w/ this comment.  It's a bit confusing that this is filtering out deleted reports, but if this is the best way you found I'm ok w/ it/

TylerSheffels commented:
1 comment otherwise good.

dannyroberts committed:
create replacement for get_document_or_404

that is more suitable for code internals; only views should call get_document_or_404
everything else should call get_document_or_not_found

dimagimon commented on corehq/apps/reports/models.py:
```diff
@@ -13,6 +13,7 @@
 from corehq.apps.domain.middleware import CCHQPRBACMiddleware
 from corehq.apps.export.models import FormQuestionSchema
 from corehq.apps.reports.display import xmlns_to_name
+from corehq.util.couch import DocumentNotFound
```
F401 'DocumentNotFound' imported but unused

dannyroberts committed:
make ConfigurableReport page 404 have some context

dimagimon commented on corehq/apps/userreports/reports/view.py:
```diff
@@ -5,7 +5,7 @@
 from django.contrib import messages
 from django.core.urlresolvers import reverse
-from django.http import HttpResponse
+from django.http import HttpResponse, Http404
```
F401 'Http404' imported but unused

dannyroberts commented:
@TylerSheffels ok how's that?

dannyroberts committed:
remove unused import
